File: python/selection.py
numbers = [42, -17, 3.14, 99, -5, 0, 23, -42, 7.5,
           16, -8.3, 101, -25, 4, 0.5, -11, 38, -9.9, 65, 12]


def selection(items: list) -> list:
    for i in range(len(items)):

        min_index = i
        aux = items[i]

        for j in range(i, len(items)):
            if items[j] < items[min_index]:
                min_index = j

        items[i] = items[min_index]
        items[min_index] = aux

    return items


print("Before selection sort:", numbers)
print("After selection sort:", selection(numbers))

File: typescript/selection.ts
const items = [42, 8, 15, 23, -3, 99, 67, 11, -45, 30, 7, 19, 88, -12, 54, 3, 26, -37, 61, 14];

export const selection = (list: number[]): number[] => {
    for (let i = 0; i < list.length; i++) {

        let min = list[i];
        let minIndex = i;
        const aux = list[i];

        for (let j = i; j < list.length; j++) {

            const current = list[j];

            if (current < min) {
                min = current;
                minIndex = j;
            }

        }

        list[i] = list[minIndex];
        list[minIndex] = aux;

    }

    return list;
}

console.log(`[selection(numbers)] -> `, selection(items))